UtopiaCity/Controllers/Sport/SportTicketController.cs

```diff
@@ -0,0 +1,181 @@
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using UtopiaCity.Models.CitizenAccount;
+using UtopiaCity.Models.Sport;
+using UtopiaCity.Services.CitizenAccount;
+using UtopiaCity.Services.Sport;
+using UtopiaCity.ViewModels.Sport;
+
+namespace UtopiaCity.Controllers.Sport
+{
+    public class SportTicketController : Controller
+    {
+        private readonly SportTicketService _sportTicketService;
+        private readonly SportComplexService _sportComplexService;
+        private readonly SportEventService _sportEventService;
+        private readonly CitizensAccountService _appUserService;
+        private readonly CitizenTaskService _citizenTaskService;
+        private readonly IMapper _mapper;
+        private readonly string _userId;
+
+        public SportTicketController(SportTicketService sportTicketService, SportComplexService sportComplexService, SportEventService sportEventService,
+            CitizensAccountService appUserService, CitizenTaskService citizenTaskService, IMapper mapper, IHttpContextAccessor httpContextAccessor)
+        {
+            _sportTicketService = sportTicketService;
+            _sportComplexService = sportComplexService;
+            _sportEventService = sportEventService;
+            _appUserService = appUserService;
+            _citizenTaskService = citizenTaskService;
+            _mapper = mapper;
+            _userId = httpContextAccessor.HttpContext.User.FindFirst(ClaimTypes.NameIdentifier).Value;
```

Comment:
Better to wrap this logic - can be changed any time - better to have 1 class to do 1 thing, being SOLID is cool